File: static/CWE-502/csharp/vulnerable.cs
// CWE-502: Deserialization of Untrusted Data
// An untrusted byte stream is deserialized with BinaryFormatter, allowing arbitrary object construction.
// Vulnerable sink: BinaryFormatter.Deserialize(stream)

using System.IO;
using System.Runtime.Serialization.Formatters.Binary;

namespace GuineaPig.Sast
{
    public class CWE_502
    {
        public object LoadObject(byte[] userInput)
        {
            using (var ms = new MemoryStream(userInput))
            {
                var formatter = new BinaryFormatter();
                return formatter.Deserialize(ms);
            }
        }
    }
}

File: static/CWE-502/java/vulnerable.java
// CWE-502: Deserialization of Untrusted Data
// Untrusted bytes are deserialized into a Java object, allowing arbitrary object instantiation.
// Vulnerable sink: new ObjectInputStream(...).readObject()

import java.io.ByteArrayInputStream;
import java.io.ObjectInputStream;

class CWE_502 {
    public Object deserialize(byte[] userInput) throws Exception {
        ByteArrayInputStream bis = new ByteArrayInputStream(userInput);
        ObjectInputStream ois = new ObjectInputStream(bis);
        return ois.readObject();
    }
}
